src/main/ruby/core/hash.rb

```diff
@@ -470,4 +470,29 @@ def to_proc
   def merge_fallback(other, &block)
     merge(Truffle::Type.coerce_to other, Hash, :to_hash, &block)
   end
+
+  def transform_values
+    if block_given?
+      h = {}
+      each_pair do |key, value|
+        h[key] = yield(value)
+      end
+      h
+    else
+      to_enum(:transform_values) { size }
+    end
+  end
+
+  def transform_values!
+    if block_given?
+      raise RuntimeError, "can't modify frozen Hash" if frozen?
```

Comment:
The `self[key]` should already perform a frozen check, isn't it enough?

Reply:
It's enough unless the Hash is empty. It could be `if empty? && frozen?`.

Reply:
I see. There is `Truffle.check_frozen` for these cases of eager frozen checks.
Could you use it?
It avoids hard-coding the frozen error message and is a bit lighter than `if frozen?`.

Reply:
Ah, nice. Done.